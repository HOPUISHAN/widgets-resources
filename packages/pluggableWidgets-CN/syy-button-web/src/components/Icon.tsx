import { createElement, ReactElement } from "react";
import classNames from "classnames";
import { WebIcon } from "mendix";
import { Icon as InternalIcon } from "@mendix/piw-utils-internal/components/web";

export interface IconProps {
    animate?: boolean;
    empty?: boolean;
    full?: boolean;
    data: WebIcon;
}

export function Icon({ animate, empty, full, data }: IconProps): ReactElement {
    let className;
    if (data && data.type === "glyph") {
        className = classNames("rating-icon", { "rating-icon-empty": empty, "rating-icon-full": full, animate });
    }
    if (data && data.type === "image") {
        className = classNames("rating-image", {
            "rating-image-empty": empty,
            "rating-image-full": full,
            animate
        });
    }
    return <InternalIcon icon={data} className={className} fallback={<div className="falltext" />} />;
}
