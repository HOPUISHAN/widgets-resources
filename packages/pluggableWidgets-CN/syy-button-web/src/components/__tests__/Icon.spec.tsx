import { createElement } from "react";
import { mount, render } from "enzyme";
import { Icon, IconProps } from "../Icon";

describe("Icon", () => {
    let defaultIconProps: IconProps;

    beforeEach(() => {
        defaultIconProps = {
            data: { type: "glyph", iconClass: "icon-class" },
            animate: true
        };
    });

    it("renders glyph icons", () => {
        const iconWrapper = render(<Icon {...defaultIconProps} />);

        expect(iconWrapper).toMatchSnapshot();
    });

    it("renders image icons", () => {
        const iconWrapper = render(<Icon {...defaultIconProps} data={{ type: "image", iconUrl: "icon.url" }} />);

        expect(iconWrapper).toMatchSnapshot();
    });

    it("renders a default icon", () => {
        const iconWrapper = render(<Icon {...defaultIconProps} data={undefined} />);

        expect(iconWrapper).toMatchSnapshot();
    });

    it("doesn't render an icon with an unknown icon data type", () => {
        const iconWrapper = render(<Icon {...defaultIconProps} data={{ type: "unknown" } as any} />);

        expect(iconWrapper).toMatchSnapshot();
    });

    it("doesn't apply an animate class to a glyph icon when animate is false", () => {
        const iconWrapper = mount(<Icon {...defaultIconProps} animate={false} />);

        expect(iconWrapper.find("span").prop("className")).not.toContain("animate");
    });

    it("doesn't apply an animate class to an image icon when animate is false", () => {
        const iconWrapper = mount(
            <Icon {...defaultIconProps} data={{ type: "image", iconUrl: "icon.url" }} animate={false} />
        );

        expect(iconWrapper.find("img").prop("className")).not.toContain("animate");
    });

    it("doesn't apply an animate class to a default icon when animate is false", () => {
        const iconWrapper = mount(<Icon {...defaultIconProps} data={undefined} animate={false} />);
        expect(iconWrapper.find("div").prop("className")).not.toContain("animate");
    });
});
