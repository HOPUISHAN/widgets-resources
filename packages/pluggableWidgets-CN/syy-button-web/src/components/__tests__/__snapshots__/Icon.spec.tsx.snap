// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon doesn't render an icon with an unknown icon data type 1`] = `
<div
  class="falltext"
/>
`;

exports[`Icon renders a default icon 1`] = `
<div
  class="falltext"
/>
`;

exports[`Icon renders glyph icons 1`] = `
<span
  aria-hidden="true"
  class="glyphicon rating-icon animate icon-class"
/>
`;

exports[`Icon renders image icons 1`] = `
<img
  alt=""
  aria-hidden="true"
  class="rating-image animate"
  src="icon.url"
/>
`;
